plugins/woocommerce/includes/admin/helper/class-wc-helper-admin.php

```diff
@@ -145,6 +160,180 @@ public static function get_featured() {
 
 		wp_send_json( $featured );
 	}
+
+	public static function check_subscriptions( $screen ) {
+		$user_id = get_current_user_id();
+		if ( ! $user_id ) {
+			return;
+		}
+
+		self::$checked_products = WC_Helper::get_checked_products();
+		if ( empty( self::$checked_products ) ) {
+			return;
+		}
+
+		self::$checked_screen_param = self::get_checked_screen_param( $screen );
```

Comment:
I'd suggest moving checks that can potentially invoke an API call to WCCOM, below other checks we do related to timestamp related checks. For an example this method can potentially trigger a call to Helper API subscription endpoint via `\WC_Helper::get_subscriptions`.

Also it might be a good idea to extract timestamp related checks to a seperate method.

Reply:
> Also it might be a good idea to extract timestamp related checks to a seperate method

Good suggestion. Updated in https://github.com/woocommerce/woocommerce/pull/47697/commits/dd23e6f325ed42f571959629d3a37ec87fea9c1f.

Reply:
> I'd suggest moving checks that can potentially invoke an API call to WCCOM, below other checks we do related to timestamp related checks

I'm not sure I understand.

Please take a look at last update — there are some renames and time-related check methods — and see whether the suggestion still apply. Basically, when current admin screen is loaded (ordered by timing), `maybe_show_product_usage_notice` does followings:

1. Bail out if not connected
1. Get notice rules (remote request to `/product-usage-notice-rules`) and cache it for 1 day
1. Get subscription state of current screen (remote request to `/subscriptions`) only if screen-and-rule-matches and cache it for 1 hour.
1. We've the rules and subscription state now; time-related throttle of the notice
1. Reaching this means modal is rendered

If we move 2) and 3) below 4), we couldn't do the check and it never reaches the remote request lines.

Reply:
> I'd suggest moving checks that can potentially invoke an API call to WCCOM, below other checks we do related to timestamp related checks

> I'm not sure I understand.

@gedex sory about that, I didn't notice `self::$product_usage_notice_rules` was used inside `self::get_current_notice_rule( $screen )`. You can ignore this.